### DISCUSSION TITLE
cannot import name 'uic' from 'PyQt5'

### DISCUSSION THREAD

**Discussion by idnovic** [samschott/maestral#378] [Q&A]
```
[idnovic@id-lappy ~]0$ maestral-gui 
Warning: PyQt5 is required to run the Maestral GUI. Run `pip install pyqt5` to install it.
Traceback (most recent call last):
  File "/home/idnovic/.local/bin/maestral", line 11, in <module>
    load_entry_point('maestral==0.2b0', 'console_scripts', 'maestral')()
  File "/home/idnovic/.local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/idnovic/.local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/idnovic/.local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/idnovic/.local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/idnovic/.local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/idnovic/.local/lib/python3.7/site-packages/maestral/console_script.py", line 30, in gui
    from maestral.gui.main import run
  File "/home/idnovic/.local/lib/python3.7/site-packages/maestral/gui/main.py", line 26, in <module>
    from maestral.gui.settings_window import SettingsWindow
  File "/home/idnovic/.local/lib/python3.7/site-packages/maestral/gui/settings_window.py", line 13, in <module>
    from PyQt5 import QtGui, QtCore, QtWidgets, uic
ImportError: cannot import name 'uic' from 'PyQt5' (/usr/lib64/python3.7/site-packages/PyQt5/__init__.py)
[idnovic@id-lappy ~]1$ pip3 install --user pyqt5
Requirement already satisfied: pyqt5 in /usr/lib64/python3.7/site-packages (5.12.2)

```

It seems I have the dependency installed but still have an import problem.
Do you have suggestions?

**Comment by samschott** [comment]
Hm, that is strange. How did you install PyQt5? And did you update it recently?

Also, does it only fail when run by Maestral, or also when you manually try to run ``from PyQt5 import uic``? 

If you are using your system's python, I would recommend installing PyQt5 through your system's package manager, e.g., ```sudo dnf install python3-PyQt5```. On Fedora 29, this works without problems.

**Comment by idnovic** [comment]
```
[idnovic@id-lappy ~]130$ python3
Python 3.7.3 (default, May 11 2019, 00:38:04) 
[GCC 9.1.1 20190503 (Red Hat 9.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from PyQt5 import uic
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'uic' from 'PyQt5' (/usr/lib64/python3.7/site-packages/PyQt5/__init__.py)
>>> 
```

Yes I am running the newest version of maestral. Also I do have PyQt5 installed (DNF). But it seems I still got a problem. I will try to reinstall it.

**Comment by idnovic** [comment]
Solved. I think some dependencies got uninstalled while I was removing unneeded software the last days. Strange that the dependency still seemed to be satisfied.

Works!

```
(1/11): qt5-qtenginio-1.6.2-24.fc30.x86_64.rpm  153 kB/s | 171 kB     00:01    
(2/11): qt5-qtconnectivity-5.12.4-1.fc30.x86_64 360 kB/s | 499 kB     00:01    
(3/11): python3-qt5-5.12.2-5.fc30.x86_64.rpm    705 kB/s | 1.1 MB     00:01    
(4/11): qt5-qtserialport-5.12.4-1.fc30.x86_64.r 112 kB/s |  64 kB     00:00    
(5/11): qt5-qtsensors-5.12.4-1.fc30.x86_64.rpm  147 kB/s | 196 kB     00:01    
(6/11): qt5-qttools-common-5.12.4-1.fc30.noarch  15 kB/s |  12 kB     00:00    
(7/11): qt5-qtlocation-5.12.4-1.fc30.x86_64.rpm 1.1 MB/s | 3.1 MB     00:02    
(8/11): qt5-qttools-libs-help-5.12.4-1.fc30.x86  87 kB/s | 142 kB     00:01    
(9/11): qt5-qtwebchannel-5.12.4-1.fc30.x86_64.r  83 kB/s |  87 kB     00:01    
(10/11): qt5-qtwebsockets-5.12.4-1.fc30.x86_64.  54 kB/s |  89 kB     00:01    
(11/11): qt5-qttools-libs-designer-5.12.4-1.fc3 729 kB/s | 2.7 MB     00:03 
```

**Comment by samschott** [comment]
Great, glad you resolved it.

**Comment by Pavancode** [comment]
how to reinstall dependencies

**Comment by morou34** [comment]
I'm having the same issue right now, and i have no idea how to install dependencies I'm working in a virtual environment, where i installed PyQt5 and PyQt5-tools, everything is perfect but PyQt5.uic fails when connecting signals

**Comment by ShanthiSenguttuvan** [comment]
I'm having the same issue right now, and i have no idea how to install dependencies. Can anyone help me?

**Comment by Microsoftened-Nair** [comment]
I found something weird with PyQt5-stubs. Pycharm shows a red wavy-underline when there is an error in a line. On uninstalling PyQt5-stubs, the red wavy underline disappears and on reinstalling PyQt5-stubs the red wavy line reappears. 
![image](https://user-images.githubusercontent.com/93315117/140608549-178b8c02-3aa0-4595-8e42-5688f7f23485.png)
![image](https://user-images.githubusercontent.com/93315117/140608619-205706cc-06d0-489f-af41-0d2b8807e71d.png)

**Reply by wahab-hac** [reply]
same issue ....

**Reply by Niyusha** [reply]
Same issue as well, uninstalling the stubs fixed the error issue. The guy I forked it from was using visual studio code instead of pycharm so perhaps the issue is somewhere in the difference those two handle the error checking for this or the venv,
